### DISCUSSION TITLE
configuring with Tailwind v4 / Alpha

### DISCUSSION THREAD

**Discussion by kunanyi** [saadeghi/daisyui#3056] [Q&A] [answered]
Has anyone managed to get this to work with the Tailwind Alpha?

I believe they are moving away from PostCSS?

https://tailwindcss.com/blog/tailwindcss-v4-alpha

The alpha has been out for a while so hoping someone has had more luck than me :)

**Comment by kunanyi** [comment]
Ah apologies there is actually a discussion already https://github.com/saadeghi/daisyui/discussions/2899

Bit sad the alpha doesn't have plugin architecture yet... hope they don't try to pin Catalyst on people

**Reply by saadeghi** [reply]
It's just alpha version for now. It's not ready for production yet.

**Comment by kunanyi** [comment]
naive question... is it possible to import daisyui as a traditional CSS file?

**Reply by saadeghi** [reply, marked as answer]
Yes, you can import the css file from node modules from `daisyui/dist/full.css`  

But this css file includes every class name. Even the ones you're not using in your project. So you might want to use an external tool like postcss-purge to remove unused class names and create a smaller CSS file for production.

Let me know if you have any questions.

**Comment by kuchta** [comment]
Seems like v4.0.0-alpha.23 already supports plugins, but it doesn't work with daisyUI since it only support single-class selectors and daisyUI uses complex ones  like media queries that will probably never be supported by the new version: https://github.com/tailwindlabs/tailwindcss/pull/14114

I assume that changing daisyUI to use just single class selectors and nesting should also work in v3, right?

**Reply by saadeghi** [reply]
Yes. The latest alpha version recently added support for plugins 🥳

I'm already working on daisyUI 5 which is going to be compatible with Tailwind CSS 4.

I will also try my best to release an update on daisyUI 4 to make sure it works with Tailwind 3. Their suggested nesting solution should work 🫡

**Reply by kuchta** [reply]
Great to hear that. Thank you for your great work on this amazing library 🙏🏻❤️

**Comment by robsontenorio** [comment]
@saadeghi  hey buddy! 

I think  we should have a side note on docs explaining how to try daisyUI alpha/beta. What do you think?

I am looking forward it to try with maryUI.

**Reply by saadeghi** [reply]
It's too soon for publish yet.  
Only a few components have been rewritten. I’ll write a blog post once all the components are added.

But if you can't wait, [install Tailwind CSS alpha](https://tailwindcss.com/blog/tailwindcss-v4-alpha)  
Then `npm i -D daisyui@alpha` and add `@import "daisyui";` to the CSS file.  
It has button, checkbox, toggle, radio, etc. I'm working on it and I will try my best to avoid unnecessary breaking changes.

**Reply by robsontenorio** [reply]
@saadeghi  Ok, I am ready 😄 
https://github.com/robsontenorio/mary/pull/623

**Comment by kuchta** [comment]
Pouya, I've noticed you are making amazing progress on the new version. Thank you for great work 🙏🏻❤️

**Comment by reidpat** [comment]
Has anyone had luck with the above method for importing daisyUI lately? I'm running into an issue with using "tailwindcss": "4.0.0-alpha.15" and "daisyui": "5.0.0-alpha.17"

Internal server error: [postcss] postcss-import: C:\Coding\animotion\node_modules\.pnpm\daisyui@5.0.0-alpha.17\node_modules\daisyui\index.js:1:9: Unknown word
  Plugin: vite:css
  File: C:/Coding/animotion/node_modules/.pnpm/daisyui@5.0.0-alpha.17/node_modules/daisyui/index.js:1:8
  1  |  import {plugin} from 'daisyui/functions/plugin.js';

**Reply by kuchta** [reply]
Unfortunatelly not. I'm using Vite with @tailwindcss/vite plugin (not postcss) and the last working version is alpha.8. Everything after that version including alpha.17 fails to build...

**Reply by saadeghi** [reply]
Tailwind added a new `@plugin` directive for plugin which allows options for plugins.
I'm working on it. Alpha version is not stable but this will be fixed in `5.0.0-alpha.18` or `5.0.0-alpha.19`

**Reply by robsontenorio** [reply]
I have a different issue after upgrading to the latest alphas

![image](https://github.com/user-attachments/assets/ba55f0b7-c25f-48bf-9ace-4520f4d84247)


![image](https://github.com/user-attachments/assets/d797cc66-7fc6-4ef0-9ea6-913fc26b5a2f)

**Reply by kuchta** [reply]
Yes, same as here 🖐🏻

**Reply by saadeghi** [reply]
@robsontenorio @kuchta @reidpat Fixed ✅  
Install `daisyui@5.0.0-alpha.20` and instead of `@import` use `@plugin`:  
https://play.tailwindcss.com/RmCP8eiTiK?file=css
```css
@import "tailwindcss";
@plugin "daisyui@5.0.0-alpha.20"
```

Thanks for testing the alpha version 💚

**update:**  
hotfix for prefers-color-scheme in `5.0.0-alpha.20`

**Reply by reidpat** [reply]
Hmm okay, no longer getting an error on import, but also not seeing elements getting styled correctly.

**Reply by saadeghi** [reply]
@reidpat can you share a small project? If it works [here](https://play.tailwindcss.com/duHReOHWpJ?file=css) it should work for you as well.

**Reply by reidpat** [reply]
This is the relevant part of the app.css
```
/* tailwind */
@import 'tailwindcss/theme';
@import 'tailwindcss/utilities';

/* @import "daisyui/dist/full.css"; */
@plugin "daisyui@5.0.0-alpha.18" {
  logs: true;
  themes: light --default, dark --prefersDark, cupcake;
}
```
```

"tailwindcss": "4.0.0-alpha.27"
"daisyui": "5.0.0-alpha.18",
```

Project works when I use the CDN import, or @import the full css.
Chances are it's my project that's being weird.

**Reply by saadeghi** [reply]
@reidpat I will try…
Both Tailwind 4 and daisyUI 5 are alpha for now, so there will be many bugs but all will be fixed until the stable version.

**Reply by kuchta** [reply]
It's working for me 👏🏻❤️

**Comment by robsontenorio** [comment]
@saadeghi The `@plugin "daisyui@5.0.0-alpha.20";` does not work, but now `@plugin "daisyui";` works... And it loads `alpha.19` as you can see on Terminal at bottom.

![image](https://github.com/user-attachments/assets/41db44c2-4294-4223-91c0-5f8447877fc6)

**Reply by kuchta** [reply]
The code was probably generated before incrementing package.json version...

**Comment by saadeghi** [comment]
I started this discussion for daisyUI 5 alpha: 
https://github.com/saadeghi/daisyui/discussions/3246

Let me if you have any feedbacks, suggestions, questions
